cmake/generic.cmake

```diff
@@ -44,7 +44,10 @@ function(cc_library TARGET_NAME)
   else()
     add_library(${TARGET_NAME} STATIC ${cc_library_SRCS})
   endif()
-  add_dependencies(${TARGET_NAME} ${cc_library_DEPS} ${external_project_dependencies})
+  if(cc_library_DEPS)
+    target_link_libraries(${TARGET_NAME} ${cc_library_DEPS})
```

Comment:
Do we need to call `target_link_librareis` when we build a library? Or, all we need here is `add_dependencies` and the linking will happen when we build shared objects or executable binaries?